src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2345,14 +2410,38 @@ QueryPushdownTaskCreate(Query *originalQuery, int shardIndex,
 	 * that the query string is generated as (...) AND (...) as opposed to
 	 * (...), (...).
 	 */
-	taskQuery->jointree->quals =
-		(Node *) make_ands_explicit((List *) taskQuery->jointree->quals);
+	if (taskQuery->jointree->quals != NULL && IsA(taskQuery->jointree->quals, List))
+	{
+		taskQuery->jointree->quals = (Node *) make_ands_explicit(
+			(List *) taskQuery->jointree->quals);
+	}
 
-	/* and generate the full query string */
-	pg_get_query_def(taskQuery, queryString);
-	ereport(DEBUG4, (errmsg("distributed statement: %s", queryString->data)));
+	subqueryTask = CreateBasicTask(jobId, taskId, taskType, NULL);
+
+	if (UpdateOrDeleteQuery(taskQuery))
+	{
+		if (HasDangerousJoinUsing(taskQuery->rtable, (Node *) taskQuery->jointree))
```

Comment:
it seems a bit weird to do this for each individual task in the planning function, rather than once in the `ModifyQuerySupported` function.